presets/ragengine/vector_store/faiss_map_store.py

```diff
@@ -0,0 +1,137 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+import numpy as np
+from typing import Any, List, Optional, cast
+
+from llama_index.core.schema import BaseNode
+from llama_index.core.bridge.pydantic import PrivateAttr
+from llama_index.vector_stores.faiss import FaissVectorStore
+from llama_index.core.vector_stores.types import (
+    VectorStoreQuery,
+    VectorStoreQueryResult,
+)
+
+class FaissVectorMapStore(FaissVectorStore):
+    # ref_doc_id_map is used to map the ref_doc_id to fiass index id
+    ref_doc_id_map: dict = {}
+    # node_id_map is used to map the faiss index id to llama_index node id
+    node_id_map: dict = {}
```

Comment:
to get around the bug in llamaindex we can use a lookup map to keep track of their node_id mappings to faiss id's

Reply:
should these go in init so its specific to each object instance?